### DISCUSSION TITLE
Are NT APIs in scope?

### DISCUSSION THREAD

**Discussion by goffrie** [microsoft/win32metadata#1071] [General] [answered]
Hi! I'd love to use the `windows-sys` crate from Rust instead of `ntapi`, but I noticed that there are many gaps in win32metadata for NT syscalls. Some examples would be filesystem things like NtQueryInformationFile, NtFlushBuffersFile, NtSetInformationFile, and so on, mostly declared in `ntifs.h`.

I noticed that in #273, @sotteson1 wrote `we excluded that file because it's used for kernel mode calls which we're not yet covering in metadata`, but many of these functions are useful from user mode too.

**Comment by kennykerr** [comment, marked as answer]
See: https://github.com/microsoft/wdkmetadata/issues/1#issuecomment-1440587834
